### DISCUSSION TITLE
How to

### DISCUSSION THREAD

**Discussion by KminekMatej** [zach-morris/plugin.program.iagl#266] [Q&A]
Hello there,
I seem not to be able to run IAGL on Kodi. My Kodi is running on Raspberry 4, Raspbian OS where Kodi was installed as a standard package. I am able to install IAGL and run games, but i have no emulator installed, which makes any game unusable.
- Kodi Menu in Game Addons doesnt display Emulators menu item at all.
- Running `apt-cache search kodi-game-*` display only available packages `kodi-game-libretro, kodi-game-libretro-dbg & kodi-game-libretro-dbgsym`
- After installing [https://github.com/zach-morris/kodi_libretro_buildbot_game_addons] the emulator menu gets visible and emulator can be installed smoothly, but no emulator can be run - after starting the game, starting any emulator crashes, displaying Unknown error in Kodi. Kodi.log displays that the process returned abnormal exit status 5.
- I tried to install retroarch as an external emulator, put correct path to executable and cfg file. Then tried to start any game, log output then looks like:
```
DEBUG <general>: IAGL:  Post processor set to NONE
INFO <general>: IAGL:  Post processing skipped for 3DConstructionKit_v1.0_1765.hdf, attempting launch
ERROR <general>: IAGL:  Launcher None is unknown, defauling to RETROPLAYER
DEBUG <general>: IAGL:  Launcher set to RETROPLAYER

```

It certainly looks like everything is against me. Is it even possible to run IAGL games on Raspbian Kodi? (v 19.4 matrix)
Thanks for any help you can provide.

**Comment by zach-morris** [comment]
Raspbian is such a pain isn't it?  Which is surprising since Retropie is such a big thing...
Anyway, you do need to install kodi-game-libretro _in addition_ to the game cores, as thats the compatibility layer that makes it all work.  Unsure exactly why there aren't any cores in the ppa from Kodi directly.

**Reply by KminekMatej** [reply]
I've got libretro.
Exactly - without any emulators available, Kodi -> Games ability simply doesnt make any sense.
Only thing that made emulators at least show up, was https://github.com/zach-morris/kodi_libretro_buildbot_game_addons - but that doesnt work. Can you tell me if Im using it correctly? (just install this addon and continue installing emulators - or should I made some compilation after install or .. ?)

```
pi@kminet:~ $ apt list --installed | grep kodi

kodi-bin/stable,now 2:19.4-3~bullseye arm64 [installed,automatic]
kodi-game-libretro-dbg/stable,now 19.0.3-3~bullseye arm64 [installed]
kodi-game-libretro-dbgsym/stable,now 19.0.3-3~bullseye arm64 [installed]
kodi-game-libretro/stable,now 19.0.3-3~bullseye arm64 [installed]
kodi-peripheral-joystick/stable,now 19.0.3-3~bullseye arm64 [installed]
kodi-vfs-libarchive/stable,now 19.2.0-3~bullseye arm64 [installed]
kodi-vfs-rar/stable,now 19.0.1-3~bullseye arm64 [installed]
kodi/stable,stable,now 2:19.4-3~bullseye all [installed]

```

**Reply by zach-morris** [reply]
Yes, so long as kodi-game-libretro (aka 'Libretro Compatibility' from the Addon Browser in Kodi) is installed, you should be able to then install the emulators from the buildbot repository.  What exactly doesn't work then when you try and launch a game?  What other errors show up in the debug log?
